pkg/providers/instancetype/types.go

```diff
@@ -335,13 +359,16 @@ func evictionThreshold(memory *resource.Quantity, storage *resource.Quantity, am
 
 func pods(ctx context.Context, info *ec2.InstanceTypeInfo, amiFamily amifamily.AMIFamily, kc *v1alpha5.KubeletConfiguration) *resource.Quantity {
 	var count int64
+	_, isWindows := amiFamily.(*amifamily.Windows)
 	switch {
+	case isWindows:
+		count = ENILimitedPods(info, amiFamily).Value()
 	case kc != nil && kc.MaxPods != nil:
```

Comment:
Are maxPods and ENILimitedPodDensity not supported by default in the default windows install?

Reply:
I don't think so.
Neither [custom networking](https://docs.aws.amazon.com/eks/latest/userguide/cni-custom-network.html) nor [IP prefixes](https://docs.aws.amazon.com/eks/latest/userguide/cni-increase-ip-addresses.html) is supported

Reply:
I just tested this by spinning up a few Windows nodes. Looks like `eniLimitedPodDensity` isn't a concept in Windows; so, a Windows node always has `--max-pods` set to 110 by default. `spec.kubeletConfiguration.maxPods` still applies because I can pass this through to `-KubeletExtraArgs` and I am able to change the pod capacity value. 

With that in mind, I think what would make the most sense is to add a `amiFamily.FeatureFlags()` that tells us whether we support `ENILimiatedPodDensity`:

```go
func pods(ctx context.Context, info *ec2.InstanceTypeInfo, amiFamily amifamily.AMIFamily, kc *v1alpha5.KubeletConfiguration) *resource.Quantity {
	var count int64
	switch {
	case kc != nil && kc.MaxPods != nil:
		count = int64(ptr.Int32Value(kc.MaxPods))
        case awssettings.FromContext(ctx).EnableENILimitedPodDensity && amiFamily.FeatureFlags().SupportsENILimitedPodDensity():
                count = ENILimitedPods(ctx, info).Value()
        default:
		count = 110
	}
	if kc != nil && ptr.Int32Value(kc.PodsPerCore) > 0 && amiFamily.FeatureFlags().PodsPerCoreEnabled {
		count = lo.Min([]int64{int64(ptr.Int32Value(kc.PodsPerCore)) * ptr.Int64Value(info.VCpuInfo.DefaultVCpus), count})
	}
	return resources.Quantity(fmt.Sprint(count))
}
```

Reply:
fixed